controllers/openshift_controller.go

```diff
@@ -943,31 +943,52 @@ func (r *KataConfigOpenShiftReconciler) processKataConfigDeleteRequest() (ctrl.R
 		// of operator functionality
 		_ = r.disablePeerPods()
 
+		// Handle podvm image deletion
+		// Since we want to declaratively reach the final state, we need to reconcile when there are errors
+		// as we want the system to give a chance of fixing the error.
+		// For cases we don't want to reconcile, ie for ImageDeletedSuccessfully and UnsupportedPodVMImageProvider
+		// we should just log the message and let the code continue without explicitly returning from the method
+
+		// Following are returned statuses:
+		// ImageDeletedSuccessfully
+		// UnsupportedPodVMImageProvider
+		// ImageDeletionFailed
+		// RequeueNeeded
+		// ImageDeletionStatusUnknown
+
 		// Handle podvm image deletion
 		status, err := ImageDelete(r.Client)
-		if status == RequeueNeeded && err == nil {
-			// Set the KataConfig status to PodVM Image Deleting
+		switch status {
+		case ImageDeletedSuccessfully:
+			r.setInProgressConditionToPodVMImageDeleted()
+			igLogger.Info("PodVM Image deleted successfully")
+
+		case UnsupportedPodVMImageProvider:
+			r.setInProgressConditionToPodVMImageUnsupportedProvider()
+			igLogger.Info("unsupported cloud provider, skipping image deletion")
+
+		case RequeueNeeded:
 			r.setInProgressConditionToPodVMImageDeleting()
-			return ctrl.Result{Requeue: true, RequeueAfter: time.Second * 15}, nil
-		} else if status == ImageDeletionFailed {
-			// Set the KataConfig status to PodVM Image Deletion Failed
+			return ctrl.Result{Requeue: true, RequeueAfter: 15 * time.Second}, err
+
+		case ImageDeletionFailed:
 			r.setInProgressConditionToPodVMImageDeletionFailed()
-			return reconcile.Result{}, err
-		} else if status == ImageDeletionStatusUnknown {
-			// Set the KataConfig status to PodVM Image Deletion Status Unknown
-			r.setInProgressConditionToPodVMImageDeletionUnknown()
+			if err != nil {
+				// We requeue only if there is an error.
+				return ctrl.Result{Requeue: true, RequeueAfter: 15 * time.Second}, err
+			}
+			// If there's no error, log and continue
+			igLogger.Info("Image deletion failed. Check logs for more details")
```

Comment:
This isn't a mere simplification of the status, it is a fix. I'd rather have it in its own commit with a bit of explanation in the commit description.